azurelinuxagent/common/osutil/default.py

```diff
@@ -257,10 +255,12 @@ def remove_legacy_firewall_rule(self, dst_ip):
                 "Unable to remove legacy firewall rule, won't try removing it again. Error: {0}".format(ustr(error)))
 
     def enable_firewall(self, dst_ip, uid):
+        # This is to send telemetry when iptable rules updated
+        is_firewall_rules_updated = False
         # If a previous attempt failed, do not retry
         global _enable_firewall  # pylint: disable=W0603
         if not _enable_firewall:
-            return False
+            return False, is_firewall_rules_updated
```

Comment:
Can you add a description to this function and describe a bit on what the 2 flags signify.

Reply:
Also, is the 2nd flag really needed? Can't we just add the telemetry inside the `enable_firewall` function?

Reply:
yes, we can but I was skeptical to add since it's utility function.

Reply:
The 2 flags seem clunky and add no major purpose except for logging. Additionally, it changes the contract of the OSUtils and might introduce additional errors if all the code paths are not verified and fixed. Could you just move the telemetry inside here? 

> yes, we can but I was skeptical to add since it's utility function.

We're anyways logging here (the multiple `logger.info`) so having telemetry doesn't make much difference